src/lib_json/json_writer.cpp

```diff
@@ -313,21 +313,26 @@ static String valueToQuotedStringN(const char* value, unsigned length) {
     // Should add a flag to allow this compatibility mode and prevent this
     // sequence from occurring.
     default: {
-      unsigned int cp = utf8ToCodepoint(c, end);
-      // don't escape non-control characters
-      // (short escape sequence are applied above)
-      if (cp < 0x80 && cp >= 0x20)
-        result += static_cast<char>(cp);
-      else if (cp < 0x10000) { // codepoint is in Basic Multilingual Plane
-        result += "\\u";
-        result += toHex16Bit(cp);
-      } else { // codepoint is not in Basic Multilingual Plane
-               // convert to surrogate pair first
-        cp -= 0x10000;
-        result += "\\u";
-        result += toHex16Bit((cp >> 10) + 0xD800);
-        result += "\\u";
-        result += toHex16Bit((cp & 0x3FF) + 0xDC00);
+      if ( emitUTF8 ) {
+        result += *c;
+      }
+      else {
+        unsigned int cp = utf8ToCodepoint(c, end);
+        // don't escape non-control characters
+        // (short escape sequence are applied above)
+        if (cp < 0x80 && cp >= 0x20)
```

Comment:
Prefer constants over magic variables, e.g.

```
const unsigned int codepoint = utf8ToCodepoint(c, end);
const int FIRST_NON_CONTROL_CODEPOINT = 0x20;
const int LAST_NON_CONTROL_CODEPOINT = 0x7F;

if (FIRST_NON_CONTROL_CODEPOINT <= codepoint && codepoint <= LAST_NON_CONTROL_CODEPOINT) {
...
```


0x80 is the maximum value, right? If you are going for non-control characters, does DEL/delete (0x7f) count as a noncontrol character?

Reply:
I didn't develop this section, just the small 'if' clause above it. I've changed it to use constants, but can't answer your question about 0x7F.